exchanges/deribit/deribit_test.go

```diff
@@ -0,0 +1,2397 @@
+	assert.NoError(t, err)
+	assert.False(t, len(instrumentInfo) == 0, "invalid instrument information found")
+	limits, err := d.GetOrderExecutionLimits(asset.Spot, spotTradablePair)
+	assert.NoErrorf(t, err, "Asset: %s Pair: %s Err: %v", asset.Spot, spotTradablePair, err)
+	assert.Falsef(t, limits.PriceStepIncrementSize != instrumentInfo[0].TickSize, "Asset: %s Pair: %s Expected: %v Got: %v", asset.Spot, spotTradablePair, instrumentInfo[0].TickSize, limits.MinimumBaseAmount)
+	assert.Falsef(t, limits.MinimumBaseAmount != instrumentInfo[0].MinimumTradeAmount, "Pair: %s Expected: %v Got: %v", spotTradablePair, instrumentInfo[0].MinimumTradeAmount, limits.MinimumBaseAmount)
+}
+
+func TestUnmarshalCancelResponse(t *testing.T) {
+	t.Parallel()
+	data := `{ "currency": "BTC", "type": "trigger", "instrument_name": "ETH-PERPETUAL", "result": [{ "web": true, "triggered": false, "trigger_price": 1628.7, "trigger": "last_price", "time_in_force": "good_til_cancelled", "stop_price": 1628.7, "replaced": false, "reduce_only": false, "price": "market_price", "post_only": false, "order_type": "stop_market", "order_state": "untriggered", "order_id": "ETH-SLTS-250756", "max_show": 100, "last_update_timestamp": 1634206091071, "label": "", "is_rebalance": false, "is_liquidation": false, "instrument_name": "ETH-PERPETUAL", "direction": "sell", "creation_timestamp": 1634206000230, "api": false, "amount": 100 }] }`
+	var resp OrderCancelationResponse
+	err := json.Unmarshal([]byte(data), &resp)
+	assert.NoError(t, err)
+}
+
+func TestGetLockedStatus(t *testing.T) {
+	t.Parallel()
+	_, err := d.GetLockedStatus(context.Background())
+	assert.NoError(t, err)
+}
+
+func TestSayHello(t *testing.T) {
+	t.Parallel()
+	_, err := d.SayHello("Sami", "")
+	assert.NoError(t, err)
+}
+
+func TestGetCancelOnDisconnect(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d)
+	_, err := d.GetCancelOnDisconnect(context.Background(), "account")
+	assert.NoError(t, err)
+	_, err = d.WsRetrieveCancelOnDisconnect("connection")
+	assert.NoError(t, err)
+}
+
+func TestDisableCancelOnDisconnect(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d, canManipulateRealOrders)
+	_, err := d.DisableCancelOnDisconnect(context.Background(), "connection")
+	assert.NoError(t, err)
+	_, err = d.WsDisableCancelOnDisconnect("connection")
+	assert.NoError(t, err)
+}
+
+func TestEnableCancelOnDisconnect(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d, canManipulateRealOrders)
+	_, err := d.EnableCancelOnDisconnect(context.Background(), "account")
+	assert.NoError(t, err)
+	_, err = d.WsEnableCancelOnDisconnect("connection")
+	assert.NoError(t, err)
+}
+
+func TestLogout(t *testing.T) {
+	t.Parallel()
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, d, canManipulateRealOrders)
+	err := d.WsLogout(true)
+	assert.NoError(t, err)
+}
+
+func TestExchangeToken(t *testing.T) {
```

Comment:
Error:      	Received unexpected error:
        	            	authenticated request failed Deribit unsuccessful HTTP status code: 400 raw response: {"jsonrpc":"2.0","error":{"message":"Invalid params","data":{"param":"refresh_token","reason":"invalid token"},"code":-32602},"testnet":true,"usIn":1708297101928835,"usOut":1708297101929126,"usDiff":291}, authenticated request failed
        	Test:       	TestExchangeToken

Are you able to retreive/generate the token yourself to make the test work?

Reply:
Nope. It's just a declaration of the method